### ISSUE TITLE
Add more resilient comparison of CATTLE_URL and Host Registration URL

### ISSUE THREAD

**Issue by superseb** [rancher/rancher#12028]
**Rancher versions:**
rancher/server: v1.6.15-rc5
rancher/agent: v1.2.10-rc3

If `Host Registration URL` is not set explicitly, the comparison added will fail and this will break setups that use automation to generate `registrationtoken` and get the command from that. This is breaking and should be reverted or fixed.

**Comment by deniseschannon** [comment]
Available with `rancher/agent:v1.2.10-rc5`

**Comment by moelsayed** [comment]
Tested with `rancher/server:v1.6.15-rc6` and `rancher/agent:v1.2.10-rc5`:
- Registration command was generated successfully when Host URL is not set.
- Proper error is generated if the Host URL is different than the Registration URL: 
```
ERROR: Configured Host Registration URL does not match given URL: CATTLE_URL=http://x.x.x.51:8080/v1 ENV_URL=http://x.x.x.50:8080/v1
ERROR: Please ensure the proper value for the Host Registration URL is set
```
